src/main/java/com/flightstats/hub/spoke/FileSpokeStore.java

```diff
@@ -40,47 +51,60 @@ public FileSpokeStore(String spokePath, int spokeTtlMinutes) {
         if (!insert("hub-startup/" + new ContentKey().toUrl(), ("" + System.currentTimeMillis()).getBytes())) {
             throw new RuntimeException("unable to create startup file");
         }
-        File file = spokeFilePathPart("hub-startup/" + new ContentKey().toUrl());
-        if (file.canExecute()) {
-            log.warn("**** Spoke file permissions may allow incomplete reads ****");
-        }
     }
 
     public boolean insert(String path, byte[] payload) {
         return insert(path, new ByteArrayInputStream(payload));
     }
 
+    @SneakyThrows
     public boolean insert(String path, InputStream input) {
         File file = spokeFilePathPart(path);
-        log.trace("insert {} {} {}", file, file.getParentFile().mkdirs(), file.canExecute());
-        try (FileOutputStream output = new FileOutputStream(file)) {
+        File tmpFile = spokeFilePathPart(SPOKE_TMP_PATH + path);
+        Stream.of(file, tmpFile).forEach(f -> f.getParentFile().mkdirs());
+        log.trace("insert {}", file);
+        filesArtificiallyLocked.add(path);
+        try (FileOutputStream output = new FileOutputStream(tmpFile)) {
             long copy = ByteStreams.copy(input, output);
-            boolean setExecutable = file.setExecutable(true);
-            log.trace("copied {} {} {}", file, copy, setExecutable);
+            log.trace("copied {} {}", file, copy);
+        } catch (IOException e) {
+            log.error("Error writing to spoke path (tmp file phase) {}", tmpFile.getPath(), e);
+            return false;
```

Comment:
In the error condition here, `filesArtificiallyLocked` is left populated.

Maybe this structure?
```
try {
   try (output) {
     ...
   }
   move
} catch {
  ...
} finally {
  ...
}
```

Reply:
yes, done.